core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -1145,6 +1145,8 @@ void rollbackOrProcessStateUpdates(
         List<InFlightState> updatedStates,
         List<PersisterStateBatch> stateBatches
     ) {
+        if (stateBatches.isEmpty() && updatedStates.isEmpty())
```

Comment:
null check maybe?

Reply:
no, since these variables are initialized at the time of their declaration, a null check is not really required.